File: Kotlin/1.kt
fun minOperation(s: String): Pair<Int, List<String>> {
    val n = s.length
    val steps = mutableListOf<String>()
    var current = ""
    
    for (i in n / 2 downTo 1) {
        if (s.substring(0, i) == s.substring(i, 2 * i)) {
            for (j in 0 until i) {
                current += s[j]
                steps.add(current)
            }
            current += current
            steps.add(current)
            current = ""
            for (j in 2 * i until n) {
                current += s[j]
                steps.add(current)
            }
            return Pair(i + 1 + (n - 2 * i), steps.toList())
        }
    }
    
    for (c in s) {
        current += c
        steps.add(current)
    }
    return Pair(n, steps.toList())
}

fun main() {
    val input = readLine()!!.trim()
    val (count, steps) = minOperation(input)
    
    print("$count = ")
    println(steps.joinToString("->"))
}

File: JS/1.js
function task1(s) {
    const n = s.length;
    const steps = [];
    let current = "";
    
    for (let i = Math.floor(n / 2); i > 0; i--) {
        if (s.substring(0, i) === s.substring(i, 2*i)) {
            for (let j = 0; j < i; j++) {
                current += s[j];
                steps.push(current);
            }
            current += current;
            steps.push(current);
            current = "";
            for (let j = 2*i; j < n; j++) {
                current += s[j];
                steps.push(current);
            }
            return [i + 1 + (n - 2*i), steps];
        }
    }
    
    for (const c of s) {
        current += c;
        steps.push(current);
    }
    return [n, steps];
}

const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('', (input) => {
    const [count, steps] = task2(input.trim());
    console.log(`${count} = ${steps.join("->")}`);
    rl.close();
});
